tests/test_selector.py

```diff
@@ -180,6 +180,9 @@ def test_re_first(self):
         self.assertEqual(sel.xpath('/ul/li/text()').re_first('\w+'), None)
         self.assertEqual(sel.xpath('/ul/li[@id="doesnt-exist"]/text()').re_first('\d'), None)
 
+        self.assertEqual(sel.re_first(r'id="(\d+)'), '1')
+        self.assertEqual(sel.re_first(r'foo'), None)
```

Comment:
This looks good to me!
For completeness sake, can we have a test for the `default` argument as well?
Thank you!!